### PR TITLE
Major bug fixes and testing done for restaurant and menu models.

### PR DESCRIPTION
**Pull request by StevenSalto**

### CODE REVIEW

File: models/restaurant.js [hunk 1]
```diff
@@ -7,25 +7,30 @@ const OM_API_KEY = process.env.OPENMENU_API_KEY
 
 class Restaurant {
     static async getMenuByRestaurantName(restaurantName, city='', postal_code=0) {
+        console.log("getMenuByRestaurantName: ", restaurantName, city, postal_code)
```
Comment (LifeBringer) [inline models/restaurant.js]: Need to remove or comment this out.

File: models/restaurant.js [hunk 2]
```diff
@@ -7,25 +7,30 @@ const OM_API_KEY = process.env.OPENMENU_API_KEY
...
+            FROM restaurants  
+                LEFT JOIN menus 
+                ON restaurants.id=menus.restaurant_id
+                LEFT JOIN items 
+                ON menus.id=items.menu_id`
         )
-    
-        if(!result.rows.length == 0) {
-            return result.rows[0]
+        console.log("db results: ", result.rows)
```
Comment (LifeBringer) [inline models/restaurant.js]: And this

File: models/restaurant.js [hunk 3]
```diff
@@ -7,25 +7,30 @@ const OM_API_KEY = process.env.OPENMENU_API_KEY
...
+        if(result.rows.length>0) {
+            return result.rows
         } else {
-            const apiRestaurantId = apiSearchForRestaurantByName(restaurantName, city, postal_code)
-            const apiRestaurant = this.apiRestaurantInfo(apiRestaurantId)
-            const dbInsertResponse = addRestaurantToDb(apiRestaurant)
+            console.log("db has no such entry, resorting to api calls: ")
```
Comment (LifeBringer) [inline models/restaurant.js]: and this

File: models/restaurant.js [hunk 4]
```diff
@@ -7,25 +7,30 @@ const OM_API_KEY = process.env.OPENMENU_API_KEY
...
+            const apiRestaurantId = await this.apiSearchForRestaurantByName(restaurantName, city, postal_code)
+            if(!apiRestaurantId) { throw new BadRequestError("No restaurant id found") }
+            const apiRestaurant = await this.apiRestaurantInfo(apiRestaurantId)
+            const dbInsertResponse = await this.addRestaurantToDb(apiRestaurant)
             return dbInsertResponse
         }
     }
     static async apiSearchForRestaurantByName(restaurantName, city, postal_code) {
-        const location = ''
+        console.log("apiSearchForRest: ", restaurantName, city, postal_code)
```
Comment (LifeBringer) [inline models/restaurant.js]: and this

File: models/restaurant.js [hunk 5]
```diff
@@ -37,21 +42,23 @@ class Restaurant {
...
-
-        const result = await axios.get(`https://openmenu.com/api/v2/location.php?key=${OM_API_KEY}&country=us&${location}=${location=='city'?city:postal_code}&s=${restaurantName}`)
+        let loc = (location=='city')?city:postal_code
+        console.log("location:", location, city, postal_code, loc, restaurantName, OM_API_KEY)
+        const result = await axios.get(`https://openmenu.com/api/v2/location.php?key=${OM_API_KEY}&country=us&${location}=${loc}&s=${restaurantName}`)
             .catch((err) => {
                 console.log(err)
             })
-        
+        console.log(result.data.response.result.restaurants[0].id)
```
Comment (LifeBringer) [inline models/restaurant.js]: and this

**Review by LifeBringer — APPROVED** [review]
Please change the 14 console.logs scattered everywhere.

**Review by StevenSalto — COMMENTED** [review]
Commented out console.logs

**Pull request merged**